src/server/middleware/session.middleware.ts

```diff
@@ -0,0 +1,53 @@
+import { Request } from 'express';
+import session from 'express-session';
+import connect from 'connect-mongodb-session';
+
+/**
+ *  representation of a express request object that includes our basic
+ *  session data.
+ */
+
+export interface SessionRequest extends Request {
+  session: {
+    cas_user: string;
+    authorizationFilterAttribute: boolean;
+  };
+}
+const MongoDBStore = connect(session);
+
+const {
+  DB_USERNAME,
+  DB_PASSWORD,
+  DB_HOSTNAME,
+  DB_PORT,
+  DB_DATABASE,
+} = process.env;
+
+const sessionStore = new MongoDBStore({
+  uri: `mongodb://${DB_HOSTNAME}:${DB_PORT}`,
+  connectionOptions: {
+    useNewUrlParser: true,
+    auth: {
+      user: DB_USERNAME,
+      password: DB_PASSWORD,
+    },
+  },
+  collection: 'sessions',
+  databaseName: DB_DATABASE,
+}, (error: Error): void => { console.error(error); });
```

Comment:
This would maybe be a good opportunity to use [the NestJS Logger](https://docs.nestjs.com/techniques/logger) to connect to CloudWatch, console.log etc.

Reply:
This is a good point, though at this point I don't believe it's 100% set that we'll be using cloudwatch for all logging. Let's take a look at some of the other logging libraries out there to see what makes the most sense for our needs and address this separately.

Reply:
Sure, but using a logging service should provide us with that flexibility - so we can swap out the underlying implementation as we see fit (perhaps we could use the built-in nest one to start with and then write our own for cloudwatch later?)